### PR TITLE
make linc script cross platform

### LINKED ISSUE #11444: Fix `yarn linc` command on Windows
It currently fails on Windows because it depends on grep:

https://github.com/facebook/react/blob/61d35ce1f3f9c0706ed3630a2650d578497ae8c2/package.json#L101

We should find a way to make it do the same in a cross platform way (e.g. from a Node script).

### PR DESCRIPTION
**Pull request by xjlim**
Fix #11444

### CODE REVIEW

File: scripts/tasks/linc.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  '--name-only',
+  '--diff-filter=ACMRTUB',
+  mergeBase,
+])
+  .trim()
+  .toString()
+  .split('\n');
+const jsFiles = changedFiles.filter(file => file.match(/.js$/g));
+try {
+  exec(eslint, jsFiles, { stdio: 'inherit' });
```
Comment (gaearon) [inline scripts/tasks/linc.js]: Maybe try using ESLint Node API instead of `exec` here?

**Comment by gaearon** [comment]
We shouldn't spawn ESLint instance for each file. This would be very slow.
Instead, use ESLint Node API to run it programmatically.

File: scripts/tasks/linc.js (hunk 1 above)
Comment (xjlim) [reply scripts/tasks/linc.js]: Thanks, will look into that.

**Comment by gaearon** [comment]
Thanks!

**Pull request merged**